prime-router/src/test/kotlin/azure/BlobAccessTests.kt

```diff
@@ -52,4 +73,237 @@ class BlobAccessTests {
+        )
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every { BlobAccess.Companion.getBlobConnection(testEnv) } returns "testconnection"
+        every { BlobAccess.Companion.uploadBlob(any(), testBytes) } returns testUrl
+
+        testEvents.forEach {
+            val result = when (it) {
+                null -> BlobAccess.uploadBody(testFormat, testBytes, testName, "")
+                Event.EventAction.CONVERT -> BlobAccess.uploadBody(testFormat, testBytes, testName, action = it)
+                else -> BlobAccess.uploadBody(testFormat, testBytes, testName, testFolder, it)
+            }
+
+            assertThat(result.format).isEqualTo(testFormat)
+            assertThat(result.blobUrl).isEqualTo(testUrl)
+            assertThat(result.digest).isEqualTo(BlobAccess.sha256Digest(testBytes))
+        }
+    }
+
+    @Test
+    fun `upload blob`() {
+        val testName = "testblob"
+        val testContainer = "testcontainer"
+        val testBytes = "testbytes".toByteArray()
+        val testEnv = "testenvvar"
+        val testUrl = "testurlname"
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every { BlobAccess.Companion.getBlobConnection(testEnv) } returns "testconnection"
+        val mockedBlobClient = mockkClass(BlobClient::class)
+        every { mockedBlobClient.upload(any<ByteArrayInputStream>(), any<Long>()) } returns(Unit)
+        every { mockedBlobClient.blobUrl } returns testUrl
+        val mockedContainerClient = mockkClass(BlobContainerClient::class)
+        every { mockedContainerClient.exists() } returns false
+        every { mockedContainerClient.create() } returns(Unit)
+        every { mockedContainerClient.getBlobClient(testName) } returns mockedBlobClient
+        val mockedServiceClient = mockkClass(BlobServiceClient::class)
+        every { mockedServiceClient.getBlobContainerClient(testContainer) } returns mockedContainerClient
+        mockkConstructor(BlobServiceClientBuilder::class)
+        every { anyConstructed<BlobServiceClientBuilder>().connectionString(any()) } answers
+            { BlobServiceClientBuilder() }
+        every { anyConstructed<BlobServiceClientBuilder>().buildClient() } returns mockedServiceClient
+        mockkConstructor(BlobClientBuilder::class)
+        every { anyConstructed<BlobClientBuilder>().connectionString(any()) } answers
+            { BlobClientBuilder() }
+        every { anyConstructed<BlobClientBuilder>().buildClient() } returns mockedBlobClient
+
+        val result = BlobAccess.uploadBlob(testName, testBytes, testContainer, testEnv)
+        val result2 = BlobAccess.uploadBlob(testName, testBytes, testContainer, testEnv)
+
+        verify(exactly = 2) { mockedBlobClient.upload(any<ByteArrayInputStream>(), testBytes.size.toLong()) }
+        assertThat(result).isEqualTo("testurlname")
+        assertThat(result2).isEqualTo("testurlname")
+
+        unmockkAll()
```

Comment:
It might be best to add a `@BeforeAll` or `@AfterAll` function to this class that calls this, it would help make sure future test cases in this file don't make the mistake of forgetting to unmock.

Reply:
I added an `@AfterEach` function to be executed in lieu of explicitly calling `unmockkAll()` after each test.

Reply:
Ah right, I got the name of that wrong, glad you figured it out!